source/GameInterface/Services/Villages/Patches/VillagePatches.cs

```diff
@@ -11,29 +21,168 @@ internal class VillagePatches
 {
     [HarmonyPatch("DailyTick")]
     [HarmonyPrefix]
-    private static bool DailyTickPrefix()
-    {
-        return false;
-    }
+    private static bool DailyTickPrefix() => ModInformation.IsServer;
+
+
+    private static readonly Func<Village, VillageStates> get_villageState = typeof(Village).GetField("_villageState",
+        BindingFlags.NonPublic | BindingFlags.Instance).BuildUntypedGetter<Village, VillageStates>();
+
 
     [HarmonyPatch(nameof(Village.VillageState), MethodType.Setter)]
     [HarmonyPrefix]
-    private static bool VillageStatePrefix()
+    private static bool VillageStatePrefix(ref Village __instance, ref VillageStates value)
     {
-        return false;
+        if(AllowedThread.IsThisThreadAllowed()) return true;
+        if (PolicyProvider.AllowOriginalCalls) return true;
+
+        if (ModInformation.IsClient) return false;
+        if (get_villageState(__instance) == value) return false;
+        
+        var message = new VillageStateChanged(__instance.StringId, (int)value);
+        MessageBroker.Instance.Publish(__instance, message);    
+        return true;
     }
 
+    public static void RunVillageStateChange(Village village, VillageStates state)
+    {
+        GameLoopRunner.RunOnMainThread(() =>
+        {
+            using(new AllowedThread())
+            {
+                village.VillageState = state;
+                village.Settlement.Party.SetLevelMaskIsDirty();
+            }
+        });
+
+    }
+
+    // Justification:
+    // At the moment looks like we dont need to handle this as the Villages that are bounded should never change.
+    // But good for more investigating soon.
+    [HarmonyPatch(nameof(Village.Bound), MethodType.Setter)]
+    [HarmonyPrefix]
+    private static bool BoundPrefix() => true;
```

Comment:
Add syncing for bound as well, some mods may update this

Reply:
I have changed my mind on this, not required. If already half way through feel free to complete otherwise it's ok to exclude this functionality